addons/project/static/src/web/activity/activity_menu_patch.js

```diff
@@ -0,0 +1,27 @@
+/* @odoo-module */
+
+import { ActivityMenu } from "@mail/web/activity/activity_menu";
+import { patch } from "@web/core/utils/patch";
+
+patch(ActivityMenu.prototype, "task", {
+
+    openActivityGroup(group, filter = "all") {
+        // fetch the data from the button otherwise fetch the ones from the parent (.o_ActivityMenuView_activityGroup).
+        const context = {};
+        if (group.model === "project.task") {
+            document.body.click();
```

Comment:
to be honest, i did not understand everything. I mostly copied the code from the crm module, since they have a similar use case.  I did try to remove the context keys, and it messed up the results so i kept them.
see : https://github.com/odoo/odoo/blob/master/addons/crm/static/src/activity_menu_patch.js